### ISSUE TITLE
chart: api server should have liveness and readiness probes

### ISSUE THREAD

**Issue by krancour** [akuity/kargo#550]

**Comment by krancour** [comment]
@devholic I am dubious of the advice in that post. This sounds like someone who ran into a specific scenario and overcompensated by deciding to avoid liveness probes altogether.

The nuanced advice in the post was basically: don't have your liveness probes depend on external things. I am dubious even of this, but giving the benefit of the doubt, I'll say there's no need for us to do that anyway.

The reason liveness probes, in some form, remain important is that failing readiness probes only stop k8s from sending traffic to that pod. They do not prompt k8s to restart the pod.

If you have a readiness probe only, it can fail forever and unless the pod exits on its own, nothing will ever cause it to restart.

Long story short, I think we still need a rudimentary liveness check.
